File: app/src/main/java/closer/vlllage/com/closer/handler/map/MapViewHandler.java
package closer.vlllage.com.closer.handler.map;

import android.app.Fragment;
import android.content.Intent;

import closer.vlllage.com.closer.pool.PoolMember;

public class MapViewHandler extends PoolMember {

    private MapSlideFragment mapSlideFragment;
    private OnRequestMapOnScreenListener onRequestMapOnScreenListener;

    public Fragment getMapFragment() {
        if (mapSlideFragment == null) {
            mapSlideFragment = new MapSlideFragment();
        }
        return mapSlideFragment;
    }

    public void onBackPressed(OnBackPressedCallback callback) {
        mapSlideFragment.post(() -> callback.onBackPressedResult(mapSlideFragment.onBackPressed()));
    }

    public void handleIntent(Intent intent) {
        mapSlideFragment.post(() -> mapSlideFragment.handleIntent(intent, onRequestMapOnScreenListener));
    }

    public MapViewHandler setOnRequestMapOnScreenListener(OnRequestMapOnScreenListener onRequestMapOnScreenListener) {
        this.onRequestMapOnScreenListener = onRequestMapOnScreenListener;
        return this;
    }

    public interface OnBackPressedCallback {
        void onBackPressedResult(boolean result);
    }

    public interface OnRequestMapOnScreenListener {
        void onRequestMapOnScreen();
    }
}

File: app/src/main/java/closer/vlllage/com/closer/handler/map/MapViewHandler.kt
package closer.vlllage.com.closer.handler.map

import android.content.Intent
import androidx.fragment.app.Fragment

import com.queatz.on.On

class MapViewHandler constructor(private val on: On) {

    private val mapSlideFragment by lazy { MapSlideFragment() }
    var onRequestMapOnScreenListener: (() -> Unit) = {}

    val mapFragment: Fragment = mapSlideFragment

    fun onBackPressed(callback: (Boolean) -> Unit) {
        mapSlideFragment.post { callback.invoke(mapSlideFragment.onBackPressed()) }
    }

    fun handleIntent(intent: Intent) {
        mapSlideFragment.post { mapSlideFragment.handleIntent(intent, onRequestMapOnScreenListener) }
    }
}
